oxidation/libparsec/crates/client_types/src/logged_core.rs

```diff
@@ -0,0 +1,53 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use once_cell::sync::OnceCell;
+use std::{
+    collections::{hash_map::DefaultHasher, HashMap},
+    hash::{Hash, Hasher},
+    sync::Mutex,
+};
+
+use libparsec_types::DeviceID;
+
+static CORE: OnceCell<Mutex<HashMap<i32, LoggedCore>>> = OnceCell::new();
+
+#[derive(Debug, Clone)]
+struct LoggedCore {
+    // TODO: Remove me when LoggedCore is implemented
+    test_device_id: DeviceID,
+}
+
+// We don't need to hash everything !
+impl Hash for LoggedCore {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.test_device_id.hash(state)
+    }
+}
+
+/// 32 bits of signature seems to be enough, people don't have that much devices
+/// Output: i32 is because javascript number is a 64 bits float
+pub fn login(test_device_id: DeviceID) -> i32 {
```

Comment:
The function should be async, especially given it will take a mutex (mutex that itself must be async of course)

Reply:
Would it be OK, to mark every function that will interact with the `database`, `filesystem` & `network` to be marked as `async` (even if it's only for testing) ?

Reply:
> Would it be OK (more like can we), to mark every function that will interact with the database, filesystem & network to be marked as async (even if it's only for testing) ?

I think you're right on this: basically all function exposed to js should be async, so better doing this by default even if async is not stricly needed for the moment because the function body is just a mock ^^